src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,437 @@
+        /// </remarks>
+        public IntPtr GetOrCreateComInterfaceForObject(object instance, CreateComInterfaceFlags flags)
+        {
+            IntPtr ptr;
+            if (!TryGetOrCreateComInterfaceForObjectInternal(this, instance, flags, out ptr))
+                throw new ArgumentException(null, nameof(instance));
+
+            return ptr;
+        }
+
+        /// <summary>
+        /// Create a COM representation of the supplied object that can be passed to a non-managed environment.
+        /// </summary>
+        /// <param name="impl">The <see cref="ComWrappers" /> implementation to use when creating the COM representation.</param>
+        /// <param name="instance">The managed object to expose outside the .NET runtime.</param>
+        /// <param name="flags">Flags used to configure the generated interface.</param>
+        /// <param name="retValue">The generated COM interface that can be passed outside the .NET runtime or IntPtr.Zero if it could not be created.</param>
+        /// <returns>Returns <c>true</c> if a COM representation could be created, <c>false</c> otherwise</returns>
+        /// <remarks>
+        /// If <paramref name="impl" /> is <c>null</c>, the global instance (if registered) will be used.
+        /// </remarks>
+        private static unsafe bool TryGetOrCreateComInterfaceForObjectInternal(ComWrappers impl, object instance, CreateComInterfaceFlags flags, out IntPtr retValue)
+        {
+            if (instance == null)
+                throw new ArgumentNullException(nameof(instance));
+
+            bool success = true;
+            object ccwValue = CCWTable.GetValue(instance, (c) =>
+            {
+                var value = CreateCCW(impl, c, flags);
+                success = value != null;
+                return value;
+            });
+            retValue = IntPtr.Zero;
+            return success;
+        }
+
+        private static unsafe object CreateCCW(ComWrappers impl, object instance, CreateComInterfaceFlags flags)
+        {
+            var vtblEntries = impl.ComputeVtables(instance, flags, out var userDefinedCount);
+            // Here I should create someing like that https://github.com/dotnet/runtime/blob/9f8aab73d93156933ae65a476204bf62c02f6537/src/coreclr/interop/comwrappers.cpp#L16
+            // Which would be saved to CCW cache.
+            // Creation of CCW is basically ManagedObjectWrapper::Create reimplementation.
+
+            // Maximum number of runtime supplied vtables.
+            Span<ComInterfaceEntry> runtimeDefinedLocal = stackalloc ComInterfaceEntry[4];
+            int runtimeDefinedCount = 0;
+
+            // Check if the caller will provide the IUnknown table.
+            if ((flags & CreateComInterfaceFlags.CallerDefinedIUnknown) == CreateComInterfaceFlags.None)
+            {
+                ComInterfaceEntry curr = runtimeDefinedLocal[runtimeDefinedCount++];
+                curr.IID = typeof(IUnknownVftbl).GUID;
```

Comment:
`Type.Guid` is just an expensive way to compute a guid. It would be better to just do `new Guid(0x00000000, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46)`.